### PR TITLE
Suppress wal lag timeout warnings right after tenant attachment

### PR DESCRIPTION
**Pull request by arpad-m**
As seen in https://github.com/neondatabase/cloud/issues/17335, during releases we can have ingest lags that are above the limits for warnings. However, such lags are part of normal pageserver startup.

Therefore, calculate a certain cooldown timestamp until which we accept lags. The heuristic is chosen to grow the later we get to fully load the tenant, and we also add 60 seconds as a grace period after that term.

### CODE REVIEW

File: pageserver/src/tenant.rs [hunk 1]
```diff
@@ -1000,11 +1003,16 @@ impl Tenant {
...
+
                 // We will time the duration of the attach phase unless this is a creation (attach will do no work)
+                let attach_start = std::time::Instant::now();
                 let attached = {
                     let _attach_timer = Some(TENANT.attach.start_timer());
                     tenant_clone.attach(preload, &ctx).await
                 };
+                let attach_duration = attach_start.elapsed();
+                let wal_lag_cooldown = attach_start + 2 * attach_duration + Duration::from_secs(60);
+                _ = tenant_clone.attach_wal_lag_cooldown.set(wal_lag_cooldown);
```
Comment (skyzh) [inline pageserver/src/tenant.rs:1015]: `tenant_clone.attach_wal_lag_cooldown.set(wal_lag_cooldown).ok();`?

**Review by skyzh — APPROVED** [review]
LGTM, but this patch seems to break backward compatibility?

**Comment by arpad-m** [comment]
> this patch seems to break backward compatibility?

yeah the test expects the warning to be present but it is still within the 60 seconds grace period. Actually it makes sense for the warning to fire in that instance.

I think the heuristic needs to be adjusted. Instead of blanket-allowing all lag warnings until a certain timestamp, it might make more sense to, until a certain timestamp, allow lag warnings that are below a certain duration. I will adjust the PR according to that.

File: pageserver/src/tenant.rs (hunk 1 above)
Comment (arpad-m) [reply pageserver/src/tenant.rs]: From what I know, the standard way to throw away results is `let _ =` and `_  =`.

**Pull request merged**